src/biogeochem/DustEmisLeung2023.F90

```diff
@@ -0,0 +1,874 @@
+         dst_emiss_coeff(p) = Cd0 * exp(-Ce * (wnd_frc_thr_slt_std - wnd_frc_thr_slt_std_min) / wnd_frc_thr_slt_std_min) ! save dust emission coefficient here for all grids
+
+         ! framentation exponent (dependent on fluid threshold)
+         frag_expt = (Ca * (wnd_frc_thr_slt_std - wnd_frc_thr_slt_std_min) / wnd_frc_thr_slt_std_min)  ! fragmentation exponent, defined in Kok et al. (2014a)
+         if (frag_expt > frag_expt_thr) then   ! set fragmentation exponent to be 3 or 5 at maximum, to avoid local AOD blowup
+            frag_expt = frag_expt_thr
+         end if
+
+         !##############################################################################################
+         !################ drag partition effect, and soil-surface friction velocity ###################
+         ! subsection on computing vegetation drag partition and hybrid drag partition factors
+         ! in Leung et al. (2023), drag partition effect is applied on the wind instead of the threshold
+         !##############################################################################################
+         ! the following comes from subr. frc_thr_rgh_fct_get
+         ! purpose: compute factor by which surface roughness increases threshold
+         !          friction velocity (currently a constant)
+
+         if (lnd_frc_mbl(p) > 0.0_r8  .AND. tlai_lu(l)<=1_r8) then
+            !if (lnd_frc_mbl(p) > 0.0_r8  .AND. ttlai(p)<=1_r8) then
+            ! vegetation drag partition equation following Gregory Okin (2008) + Caroline Pierre et al. (2014)
+            !lai(p) = tlai_lu(l)+0.1_r8       ! LAI+SAI averaged to landunit level; the equation is undefined at lai=0, and LAI in CTSM has some zeros over deserts, so we add in a small number.
+            !lai(p) = ttlai(p) + lai0_Okin     ! ttlai = tlai+tsai. Okin-Pierre's equation is undefined at lai=0, and LAI in CTSM has some zeros over deserts, so we add in a small number. On 26 Feb 2024, dmleung changed from tlai_lu(l) to ttlai(p)
+            !if (lai(p) > 1_r8) then
+            !   lai(p)  = 1_r8   ! setting LAI = 1 to be a max value (since K_length goes to negative when LAI>1)
+            !end if              !
+
+            !if (ttlai(p) + vai0_Okin <= 1_r8) then
+            !   vai_Okin(p) = ttlai(p) + vai0_Okin     ! ttlai = vai = tlai+tsai. Okin-Pierre's equation is undefined at vai=0, and VAI in CTSM has some zeros over deserts, so we add in a small number. On 26 Feb 2024, dmleung changed from tlai_lu(l) to ttlai(p)
+            !end if                                    ! In the Okin-Pierre formulation, VAI has to be 0 < VAI <= 1.
+
+            vai_Okin(p) = tlai_lu(l)+vai0_Okin       ! LAI+SAI averaged to landunit level; the equation is undefined at lai=0, and LAI in CTSM has some zeros over deserts, so we add in a small number.
+            if (vai_Okin(p) > 1_r8) then
```

Comment:
Again should the 1_r8 here and below be vai_mbl_thr?

Reply:
I think I'll leave it alone there is some description of the limits of this already in comments.